.github/workflows/deploy.yml

```
@@ -0,0 +1,106 @@
+name: Continuous deployment
+
+on:
+  push:
+    branches:
+      - master
+
+env:
+  RUST_BACKTRACE: 1
+  PRUSTI_ASSERT_TIMEOUT: 60000
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ macos-latest, ubuntu-latest, windows-latest ]
+      # Don't cancel when one matrix job fails, so we can compare to the others.
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Setup Python 3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1.0.6
+      - name: Set up the environment
+        run: python x.py setup
+      - name: Build with cargo --release
+        run: python x.py build --all --verbose --release
+      - name: Run "quick" cargo tests
+        run: python x.py test --all --verbose quick
+      - name: Upload Prusti artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: prusti-release-${{ matrix.os }}
+          if-no-files-found: error
+          path: |
+            rust-toolchain
+            viper_tools/
+            target/release/prusti-driver
+            target/release/prusti-server-driver
+            target/release/prusti-server
+            target/release/prusti-rustc
+            target/release/cargo-prusti
+            target/release/libprusti_contracts.rlib
+            target/release/libprusti_contracts_internal.*
+            !target/release/libprusti_contracts_internal.d
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Download all Prusti artifacts
+        uses: actions/download-artifact@v2
+      - name: Package Prusti artifacts
+        shell: bash
+        run: |
+          for os in ubuntu windows macos
+          do
+            echo "Package Prusti artifact for $os"
+            cd prusti-release-$os-latest
```

```suggestion
            cd prusti-release-$os
```

IIUC this has to be the same as the ones from upload artifact?

---

Yes, I think it has to be the same

---

Right, but doesn't this have to match line 38? (Which also doesn't have the suffix?)

---

> Right, but doesn't this have to match line 38? 

Yes.

> (Which also doesn't have the suffix?)

`matrix.os` iterates over line 16, so the suffix should be there.